components/fxa-client/src/internal/close_tabs.rs

```diff
@@ -19,19 +24,70 @@ impl FirefoxAccount {
             .iter()
             .find(|d| d.id == target_device_id)
             .ok_or_else(|| Error::UnknownTargetDevice(target_device_id.to_owned()))?;
-        let (payload, sent_telemetry) =
-            CloseTabsPayload::with_urls(urls.iter().map(|url| url.as_ref().to_owned()).collect());
-        let oldsync_key = self.get_scoped_key(scopes::OLD_SYNC)?;
-        let command_payload =
-            encrypt_command(oldsync_key, target, close_tabs::COMMAND_NAME, &payload)?;
-        self.invoke_command(
-            close_tabs::COMMAND_NAME,
-            target,
-            &command_payload,
-            Some(close_tabs::COMMAND_TTL),
-        )?;
-        self.telemetry.record_command_sent(sent_telemetry);
-        Ok(())
+
+        let sent_telemetry = telemetry::SentCommand::for_close_tabs();
+        let mut urls_to_retry = Vec::new();
+
+        // Sort the URLs shortest to longest, so that we can at least make
+        // some forward progress, even if there's an oversized URL at the
+        // end that won't fit into a single command.
+        let mut urls: Vec<_> = urls.iter().map(|url| url.as_ref().to_owned()).collect();
+        urls.sort_unstable_by_key(String::len);
+
+        while !urls.is_empty() {
+            // If we were asked to close more URLs than will fit in a
+            // single command, chunk the URLs into multiple commands,
+            // packing as many as we can into each. Do this until we've either
+            // drained and packed all the URLs, or we see an oversized URL
+            // that won't fit into a single command.
+            let chunk = match payload_support::try_fit_items(
+                &urls,
+                // This can be simplified to `COMMAND_MAX_PAYLOAD_SIZE.get()`
+                // once we bump our MSRV to 1.73.0.
+                COMMAND_MAX_PAYLOAD_SIZE.with(Cell::get),
```

Comment:
The `LocalKey<Cell<T>>` convenience methods were [stabilized in 1.73.0](https://blog.rust-lang.org/2023/10/05/Rust-1.73.0.html#thread-local-initialization).

m-c current MSRV is [1.76.0](https://searchfox.org/mozilla-central/rev/0b1d02b2cb5736511139cf0e40b318273e825899/python/mozboot/mozboot/util.py#14); ours is [1.70.0](https://github.com/mozilla/application-services/blob/16fdc3b6fee4c5404e77f7034822f5b5d9c464e2/.circleci/config.yml#L16). I wonder if we could bump ours?

Reply:
Our MSRV is now 1.76.0, as of #6301, so we can use those convenience methods now! 💃🏻